openedx/core/djangoapps/safe_sessions/tests/test_utils.py

```diff
@@ -118,14 +130,20 @@ def assert_request_user_mismatch(self, user_at_request, user_at_response):
             yield
 
     @contextmanager
-    def assert_session_user_mismatch(self, user_at_request, user_in_session):
-        """
-        Asserts that the logger was called when request.user at request
-        time doesn't match the request.user at response time.
-        """
-        with self.assert_logged(
-            "SafeCookieData user at request '{}' does not match user in session: '{}'".format(
-                user_at_request, user_in_session
-            ),
+    def assert_request_and_session_user_mismatch(self, user_at_request, user):
```

Comment:
Yeah.  I like your latest suggestion here.  It makes it easier to understand.  I also suggest removing the docstrings as they aren't adding any additional value, but only if you rename the methods so they are clearer: `assert_logged_for_request_user_mismatch` and `assert_logged_for_session_user_mismatch`.